R/crunchgeo.R

```diff
@@ -0,0 +1,76 @@
+# Here's a good place to put your top-level package documentation
+
+.onAttach <- function (lib, pkgname="crunchgeo") {
+    ## Put stuff here you want to run when your package is loaded
+    invisible()
+}
+
+#' @rdname fetchGeoFile
+setGeneric("fetchGeoFile", function (geography, ...) standardGeneric("fetchGeoFile"))
+
+
+#' Fetch a geo- or topo-json hosted by Crunch
+#'
+#' @param geography a `CrunchGeography` object, generally obtained by using the `geo`
+#'  function on a dataset variable that has geography data specified.
+#' @param ... additional options to pass to `geojson_read` or `topojson_read`
+#'
+#' @examples
+#' \dontrun{
+#' geo_data <- fetchGeoFile(geo(ds$country))
+#' }
+#'
+#' @importFrom tools file_ext
+#' @importFrom httpcache halt
+#' @importFrom geojsonio geojson_read topojson_read
+#' @import crunch
+#' @rdname fetchGeoFile
+#' @export
+setMethod("fetchGeoFile", "CrunchGeography", function(geography, ...){
```

Comment:
You may want to add methods for other classes, such as

* CrunchVariable: function (geography) fetchGeoFile(geo(geography))
* ANY: halt("fetchGeoFile not defined for class X") or whatever (something nicer than the coarse method dispatch error)